packages/core/core-flows/src/payment/steps/capture-payment.ts

```diff
@@ -36,6 +36,26 @@ export const capturePaymentStep = createStep(
 
     const payment = await paymentModule.capturePayment(input)
 
-    return new StepResponse(payment)
+    return new StepResponse(payment, payment)
+  },
+  async (payment, { container }) => {
+    if (!payment) {
+      return
+    }
+
+    const paymentModule = container.resolve<IPaymentModuleService>(
+      Modules.PAYMENT
+    )
+
+    // Every time we capture a payment, we create a new capture. We want to delete the newset capture if we capture a payment again.
+    const latestCapture = (payment.captures ?? []).sort((a, b) => {
+      return new Date(b.created_at).getTime() - new Date(a.created_at).getTime()
+    })[0]
+
+    if (!latestCapture) {
+      return
+    }
+
+    await paymentModule.deleteCaptures([latestCapture.id])
```

Comment:
thought/question: If the payment has been captured at the third-party provider, and the compensation flow kicks in, we are deleting the Medusa captures but not the third-party capture (likely because we can't). 

Could this not leave the application in a state of inconsistency, for example, a captured payment in Stripe without an associated record in Medusa. I would have the same concern for refunds. 

I believe this is why we originally did not add compensation functions to these steps.

Reply:
I guess it's about what sort of "consistency" you want, but I didn't think about it like that, I think you are right. I'll add comments for the next naive person to not add it again 😄